util/fibers/detail/fiber_interface.cc

```diff
@@ -274,6 +274,23 @@ ctx::fiber_context FiberInterface::SwitchTo() {
   });
 }
 
+void FiberInterface::SwitchToAndExecute(std::function<void()> fn) {
+  FiberInterface* prev = SwitchSetup();
+
+  // pass pointer to the context that resumes `this`
+  std::move(entry_).resume_with([prev, fn = std::move(fn)](ctx::fiber_context&& c) {
```

Comment:
this is where the bug was. the original line was `entry_ = std::move(....`.
`entry_ = ` part was a weird typo.

Reply:
oh wow!